File: AtCoder_Beginners_Selection/ABC049C/main.nim
import math
import strutils
proc main =
    var S: string = stdin.readLine()

    # input check
    assert(1 <= len(S) and len(S) <= 10^5)

    if S.replace("eraser").replace("erase").replace("dreamer").replace("dream") == "":
        echo "YES"
        return
    echo "NO"

when isMainModule:
    main()

File: AtCoder_Beginners_Selection/ABC049C/main.py

def main():
    S: str = input()
    T: str = ""
    pattern: list[str] = ["dream", "dreamer", "erase", "eraser"]
    # input check
    assert(1 <= len(S) <= pow(10, 5))

    S=S.replace("eraser", "")
    S=S.replace("erase", "")
    S=S.replace("dreamer", "")
    S=S.replace("dream", "")
    if S == "":
        print("YES")
        return
    print("NO")

if __name__ == "__main__":
    main()
